tests/sharding/test_vmc_handler.py

```diff
@@ -135,26 +147,131 @@ def deploy_initiating_contracts(vmc_handler, privkey):
         vmc_handler.config['GAS_PRICE'],
     )
     for tx in txs:
-        send_raw_transaction(vmc_handler, tx)
+        tx_hash = send_raw_transaction(vmc_handler, tx)
         mine(vmc_handler, 1)
+        assert w3.eth.getTransactionReceipt(tx_hash) is not None
     logger.debug(
         'deploy_initiating_contracts: vmc_tx_hash=%s',
         w3.eth.getTransactionReceipt(encode_hex(txs[-1].hash)),
     )
 
 
-def import_key(vmc_handler, privkey):
+def deploy_vmc(vmc_handler):
+    # test the deployment of vmc ######################################
+    # deploy vmc if it is not deployed yet.
+    if not is_vmc_deployed(vmc_handler):
```

Comment:
I'm seeing this pattern here and in `import_key` and I believe we can replace it with a `pytest.fixture`

It looks like `tests.sharding.fixtures.vmc` should be renamed to `vmc_handler`

Then define:

```python
@pytest.fixture
def primary_key(vmc_handler):
    # the body of the current `import_key` without the `try/except` logic.

@pytest.fixture
def vmc(vmc_handler, primary_key):
    # the body of `deploy_vmc` without the `if` block.
```

Since we know pytest will only run these *once* for each tests we should get strong guarantees that they won't cause errors.  The one exception *may* be the importing of the key as I'm not 100% sure what the logic behind the scenes is for how imported accounts are managed and if they are reset on each test run (I think they should be).

Reply:
Will do it. Thank you!:)